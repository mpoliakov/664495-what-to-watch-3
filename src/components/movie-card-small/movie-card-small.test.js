import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {BrowserRouter} from 'react-router-dom';
import configureStore from 'redux-mock-store';
import MovieCardSmall from './movie-card-small.jsx';

const mockStore = configureStore([]);

it(`<SmallMovieCard/> is rendered correctly`, () => {
  const film = {
    id: 1,
    name: `Alien`,
    previewImage: `img/alien.jpg`,
    previewVideoLink: `https://movies.com/alien.ogv.360p.webm`,
  };

  const store = mockStore({});

  const onMouseEnter = jest.fn();
  const onMouseLeave = jest.fn();

  const component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <MovieCardSmall film={film} renderVideoPlayer={() => {
          }} onMouseEnter={onMouseEnter} onMouseLeave={onMouseLeave}/>
        </BrowserRouter>
      </Provider>
  ).toJSON();

  expect(component).toMatchSnapshot();
});
