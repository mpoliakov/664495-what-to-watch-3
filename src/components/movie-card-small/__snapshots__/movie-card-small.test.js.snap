// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCardSmall/> is rendered correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[MockFunction]}
  onMouseLeave={[MockFunction]}
>
  <div
    className="small-movie-card__image"
  />
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      Alien
    </a>
  </h3>
</article>
`;

exports[`<SmallMovieCard/> is rendered correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[MockFunction]}
  onMouseLeave={[MockFunction]}
>
  <div
    className="small-movie-card__image"
  />
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      Alien
    </a>
  </h3>
</article>
`;
